cmd/cluster_addon.go

```diff
@@ -51,8 +51,8 @@ func validateAddonSubCommand(cmd *cobra.Command, args []string) error {
 	if idx == -1 {
 		return fmt.Errorf("cluster '%s' not found", name)
 	}
-	if len(args) != 1 {
-		return errors.New("exactly one argument expected")
+	if len(args) < 1 {
```

Comment:
One question, since is not clear to me the expected behaviour.

Can I invoke this add-on with more than one argument and the first one should be the script, eg:
```
hetzner-kube cluster addon install --name script-runner /PATH/TO/MY/SCRIPT FIRST_ARG SECOND_ARG THIRD_ARG
```
or I need to run it with only one argument that indicate the path script file like:
```
hetzner-kube cluster addon install --name script-runner /PATH/TO/MY/SCRIPT
```
In the first case I expect that extra args are propagated to the script on the server, in the second case I expect to have an error if there are some extra args that will be ignored.

Reply:
No, extra args will not be sent to the script. The correct call should be 
```
hetzner-kube cluster addon install --name script-runner /PATH/TO/MY/SCRIPT
```
And the script will be called with node group and cluster info as args, ex:
```
/tmp/script.sh worker '{...}'
```

Reply:
Than maybe is better that we check that `if len(args) != 2 { }` and if not return an error.

Reply:
I think since this validation is for all addons, we do not want to restrict it.

Reply:
Right, maybe we should add a new method `ArgsValidation` to the `Addon` interface and delegate each plugin to validate data they need (I mean in a different PR), WDYT?

Reply:
IMHO I would leave it up to the addon to decide it without any restriction.